sw/device/silicon_creator/rom/e2e/rom_e2e_bootstrap_rma_test.c

```diff
@@ -0,0 +1,73 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/rstmgr_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+static const char *reset_reason_to_string(dif_rstmgr_reset_info_t reset_info) {
+  switch (reset_info) {
+    case kDifRstmgrResetInfoPor:
+      return "Por";
+    case kDifRstmgrResetInfoLowPowerExit:
+      return "LowPowerExit";
+    case kDifRstmgrResetInfoSw:
+      return "Sw";
+    case kDifRstmgrResetInfoHwReq:
+      return "Hw";
+    case kDifRstmgrResetInfoSysRstCtrl:
+      return "SysRstCtrl";
+    case kDifRstmgrResetInfoWatchdog:
+      return "Watchdog";
+    case kDifRstmgrResetInfoPowerUnstable:
+      return "PowerUnstable";
+    case kDifRstmgrResetInfoEscalation:
+      return "Escalation";
+    case kDifRstmgrResetInfoNdm:
+      return "Ndm";
+  }
+}
+
+// This function dumps the reset reasons contained within `bitfield` to enable
+// human-readable regular expressions in the test harness. The reasons are
+// printed in a deterministic order, separated by commas.
```

Comment:
nit: I wonder if this complexity is worth having slightly simpler regexps on the host side. Have you considered printing the number of reset reasons followed by each reason on a line? Another idea that we could explore is to log whatever you want freely but match the hex representation of reset reasons.

Reply:
OK, I've gone the route of just logging a hex value in the C code, which avoids the messy string construction. I've basically traded that complexity for some additional bindgen glue. Since this is a somewhat significant change, I'll ask you take take another look before I merge this :)